geomesa-features/geomesa-feature-avro/src/main/scala/org/locationtech/geomesa/features/avro/serialization/SimpleFeatureDatumReader.scala

```diff
@@ -0,0 +1,109 @@
+/***********************************************************************
+ * Copyright (c) 2013-2022 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.features.avro
+package serialization
+
+import org.apache.avro.Schema
+import org.apache.avro.io.{DatumReader, Decoder}
+import org.locationtech.geomesa.features.ScalaSimpleFeature
+import org.locationtech.geomesa.features.avro.serialization.SimpleFeatureDatumReader.VersionedFields
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+/**
+ * Avro reader for simple features. Due to avro lifecycles, the following methods must be
+ * called in order before use:
+ *
+ * 1) setSchema (if available)
+ * 2) setFeatureType
+ */
+class SimpleFeatureDatumReader extends DatumReader[SimpleFeature] {
```

Comment:
Are the SimpleFeatureDatumReader/Writers covered by tests? Do they need to be?

Reply:
i think they're covered by the older tests but maybe i should double check that.

Reply:
They're covered by the KDS tests and the AvroDataFileTest but I added a new one just for the reader/writer directly